kogito-editors-js/packages/boxed-expression-component/src/components/DecisionTableExpression/DecisionTableExpression.tsx

```diff
@@ -106,36 +107,37 @@ export function DecisionTableExpression(decisionTable: PropsWithChildren<Decisio
     return editColumnLabel;
   }, [i18n.editClause.input, i18n.editClause.output]);
 
-  const columns = useMemo(() => {
-    const inputColumns = (decisionTable.input ?? [{ name: "input-1", dataType: DataType.Undefined }])?.map(
-      (inputClause) =>
+  const columns = useMemo<ColumnInstance[]>(() => {
+    const inputColumns = _.map(
```

Comment:
I have several reasons for preferring Lodash functions over native ones, for example:
- Lodash functions are null-safe: no need to check if the input is null/undefined;
- Performance: https://www.measurethat.net/Benchmarks/Show/10250/0/lodash-map-vs-native-map-v2
- My impression is that Lodash support to types is better.

Reply:
TypeScript is responsible to handle types and the `map` function for iterating over the array. If the array is null/undefined you can chain it with `?` for null-checks, or in case you want to remove null elements, you can use a `filter` function.

The native implementation uses the fluid pattern, where you can chain methods easily, which is the pattern used by most TypeScript/React projects, kogito-tooling included. Also, if you wish you can use the same pattern with the `chain` method. Like so ```_.chain(array).map(...).filter(...).reduce(...).value()```

Regarding the types, I have a simple example where it's not true. In this example, lodash can't type the `key` and `value`.
```
const map = new Map<string, number>();
_.map(Array.from(map.entries()), ([key, value]) => console.log(key, value));

Array.from(map.entries()).map(([key, value]) => console.log(key, value));
```

I'm fine to use lodash, but please use the `chain` method to maintain the code consistency with other projects we have.

Reply:
@ljmotta, thanks for your valuable feedback.
TBH, I can't entirely agree with you for all points you mentioned.
For instance, the null safety provided by lodash is about the input argument itself and not the nested properties (for that case, the optional chaining can help, obviously).
```
_.map(null)
> []
_.map(undefined)
> []
```
About the fluid pattern, be aware that every single chained method `.map`, `.filter`, `.reduce` causes a complete traverse of the whole collection (complexity `O(n)`), and it should be avoided when the collection is huge (like `rows` and `columns` can be.
Regarding the types, it was just my sensation: with Lodash functions, there was no need to specify argument types for the wrapped functions, but that's definitely not a problem.

That's just my point of view. I respect yours, and I'm going to accomplish your feedback because I want to unblock this PR.

Thanks again!